### PR TITLE
[camera_android] Upgrading roboelectric from 4.5 to 4.10.3

### PR DESCRIPTION
**Pull request by gmackall**
https://github.com/flutter/flutter/issues/119752

Upgrades to allow configurations of sdk >= 31 (4.8.1 was chosen as it is the most common version used in other packages).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera_android/android/build.gradle [hunk 1]
```diff
@@ -51,6 +51,7 @@ android {
         unitTests.includeAndroidResources = true
         unitTests.returnDefaultValues = true
         unitTests.all {
+            jvmArgs "-Xmx1g"
```
Comment (camsim99) [inline packages/camera/camera_android/android/build.gradle:54]: Bumped to fix OOM errors.

File: packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/media/MediaRecorderBuilder.java [hunk 2]
```diff
@@ -92,7 +92,6 @@ public MediaRecorder build() throws IOException, NullPointerException, IndexOutO
       mediaRecorder.setVideoEncodingBitRate(videoProfile.getBitrate());
       mediaRecorder.setVideoFrameRate(videoProfile.getFrameRate());
       mediaRecorder.setVideoSize(videoProfile.getWidth(), videoProfile.getHeight());
-      mediaRecorder.setVideoSize(videoProfile.getWidth(), videoProfile.getHeight());
```
Comment (camsim99) [inline packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/media/MediaRecorderBuilder.java:95]: Unrelated fix: remove duplicate line

File: packages/camera/camera_android/android/src/test/java/io/flutter/plugins/camera/features/resolution/ResolutionFeatureTest.java [hunk 3]
```diff
@@ -91,8 +91,9 @@ public void beforeLegacy() {
   public void before() {
     mockProfileLow = mock(EncoderProfiles.class);
     EncoderProfiles mockProfile = mock(EncoderProfiles.class);
-    EncoderProfiles.VideoProfile mockVideoProfile = mock(EncoderProfiles.VideoProfile.class);
-    List<EncoderProfiles.VideoProfile> mockVideoProfilesList = List.of(mockVideoProfile);
+    List<EncoderProfiles.VideoProfile> mockVideoProfilesList =
+        new ArrayList<EncoderProfiles.VideoProfile>();
+    mockVideoProfilesList.add(null);
```
Comment (camsim99) [inline packages/camera/camera_android/android/src/test/java/io/flutter/plugins/camera/features/resolution/ResolutionFeatureTest.java:96]: The `EncoderProfiles.VideoProfile` class is final static so it cannot be mocked. It's not actually tested in this class so I just set it to null so that the tests run properly.

File: packages/camera/camera_android/android/src/test/java/io/flutter/plugins/camera/features/resolution/ResolutionFeatureTest.java [hunk 4]
```diff
@@ -386,7 +385,7 @@ public void computeBestPreviewSize_shouldUseLegacyBehaviorWhenEncoderProfilesNul
   @Config(minSdk = 31)
   @Test
   public void resolutionFeatureShouldUseLegacyBehaviorWhenEncoderProfilesNull() {
-    beforeLegacy();
+    before();
```
Comment (camsim99) [inline packages/camera/camera_android/android/src/test/java/io/flutter/plugins/camera/features/resolution/ResolutionFeatureTest.java:388]: This tests non-legacy code so this was incorrect I think.

**Comment by camsim99** [comment]
This is a test-only change except for an unrelated fix, so I added a `NEXT` entry. Let me know if I need to actually do a version bump.

File: packages/camera/camera_android/android/gradle.properties [hunk 5]
```diff
@@ -0,0 +1 @@
+org.gradle.jvmargs=-Xmx16Gheheheheheh
```
Comment (stuartmorgan) [inline packages/camera/camera_android/android/gradle.properties:1]: What's the `heheheheheh`?

**Review by stuartmorgan — APPROVED** [review]
LGTM with one question. I'm fine with not releasing the duplicate line removal since while it technically changes runtime behavior, it's hard to imagine it being anything but a no-op in effect.

File: packages/camera/camera_android/android/gradle.properties (hunk 5 above)
Comment (camsim99) [reply packages/camera/camera_android/android/gradle.properties]: This was a test to see if I could change the heap size this way. It didn't impact anything and I forgot to remove it -- oops!
